src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +163,37 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public async Task<string> TryGetImageName(
+            string logUri,
+            Regex imageNameRegex,
+            Action<Exception> exceptionHandler,
+            CancellationToken cancellationToken)
+        {
+            return await _retry.RetryAsync(async () =>
```

Comment:
I think there's a potential issue here. If the cancellation token expires exponential retry will try to retry this and hit it again.